anthropic-bedrock-model-provider/main.py

```diff
@@ -47,10 +50,36 @@ async def log_body(request: Request, call_next):
 async def get_root():
     return uri
 
-
+# https://docs.anthropic.com/en/api/claude-on-amazon-bedrock#list-available-models
 @app.get("/v1/models")
 async def list_models() -> JSONResponse:
-    return await claude3_provider_common.list_models(client)
+    try:
+        bedrock = boto3.client(service_name="bedrock")
+        response = bedrock.list_foundation_models(byProvider="anthropic")
+    except Exception as e:
+        return JSONResponse(content={"error": f"Failed to list models - bedrock client error: {e}"}, status_code=500)
+    try:
+        if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+            log(f"Failed to list models: {response}")
+            return JSONResponse(content={"error": f"Failed to list models - unexpected status code: {response}"},
+                                status_code=response["ReponseMetadata"]["HTTPStatusCode"])
+    except KeyError as e:
+        return JSONResponse(content={"error": f"Failed to list models - bad response: {e}"}, status_code=500)
+
+    try:
+        models = []
+        for model in response["modelSummaries"]:
+            if model["modelLifecycle"]["status"] != "ACTIVE" or "PROVISIONED" not in model["inferenceTypesSupported"]:
```

Comment:
`"PROVISIONED" not in model["inferenceTypesSupported"]` seems suspicious to me. I would expect `PROVISIONED` to be an inference type. If this is correct, then please ignore. Just wanted to call it out.

Reply:
There's PROVISIONED and ONDEMAND, but only the former worked properly without extra AWS configuration when I tested it, so I just ignore all others here.